packages/harmony/src/components/button/FilterButton/FilterButton.tsx

```diff
@@ -0,0 +1,218 @@
+          : color.text.default,
+      gap: spacing.xs,
+      fontSize: typography.size.s,
+      fontWeight: typography.weight.medium,
+      lineHeight: typography.lineHeight.s,
+
+      '&:hover': {
+        border: `1px solid ${color.neutral.n800}`,
+        transform: 'none'
+      },
+
+      '&:active': {
+        ...activeStyle,
+        transform: 'none'
+      },
+
+      ...(size === FilterButtonSize.SMALL ? smallStyles : defaultStyles),
+      ...(isOpen ? activeStyle : {}),
+      ...(variant === FilterButtonType.FILL_CONTAINER && selection !== null
+        ? fillContainerStyles
+        : {})
+    }
+
+    const iconCss =
+      size === FilterButtonSize.SMALL ? smallIconStyles : defaultIconStyles
+
+    // Popup Styles
+    const optionCss: CSSObject = {
+      background: 'transparent',
+      border: 'none',
+      color: color.text.default,
+      fontWeight: typography.weight.medium,
+      gap: spacing.s,
+      paddingLeft: spacing.m,
+      paddingRight: spacing.m,
+      paddingTop: spacing.s,
+      paddingBottom: spacing.s,
+      width: '100%',
+      borderRadius: cornerRadius.s,
+      justifyContent: 'flex-start',
+
+      '&:hover': {
+        transform: 'none',
+        backgroundColor: color.secondary.s300,
+        color: color.special.white
+      },
+
+      '&:active': {
+        transform: 'none'
+      }
+    }
+    const optionIconCss: CSSObject = {
+      width: spacing.unit4,
+      height: spacing.unit4
+    }
+
+    const handleButtonClick = useCallback(() => {
+      if (variant === FilterButtonType.FILL_CONTAINER && selection !== null) {
+        setSelection(null)
+      } else {
+        setIsOpen((isOpen: boolean) => !isOpen)
+      }
+    }, [selection, variant, setIsOpen, setSelection])
+
+    const handleOptionSelect = useCallback(
+      (option: FilterButtonOption) => {
+        setSelection(option)
+        onSelect?.(option)
+      },
+      [onSelect]
+    )
+
+    const anchorRef = useRef<HTMLButtonElement>(null)
+
+    return (
+      <BaseButton
+        ref={ref || anchorRef}
+        styles={{
+          button: buttonCss,
+          icon: iconCss
+        }}
+        onClick={handleButtonClick}
+        iconRight={
+          variant === FilterButtonType.FILL_CONTAINER && selection !== null
+            ? IconCloseAlt
+            : iconRight
+        }
+        aria-haspopup='listbox'
+        aria-expanded={isOpen}
+      >
+        {selection?.label ?? label}
+        <Popup
+          anchorRef={(ref as RefObject<HTMLElement>) || anchorRef}
+          isVisible={isOpen}
+          onClose={() => setIsOpen(false)}
+          anchorOrigin={popupAnchorOrigin}
+          transformOrigin={popupTransformOrigin}
+        >
+          <Paper mt='s' border='strong' shadow='far'>
+            <Box p='s'>
+              <Flex
+                direction='column'
+                alignItems='flex-start'
+                justifyContent='center'
+                role='listbox'
```

Comment:
selects should typically have a label, so maybe an aria-labelledby with the label above? ^ you should be able to confirm your a11y changes by adding a test to storybook. you can reference https://github.com/AudiusProject/audius-protocol/blob/55f295726daa5b090e018099ba2e6c40d9ff0e2b/packages/harmony/src/components/button/Button/Button.stories.tsx#L177

Reply:
this is awesome